renumber Peek children, the sequel

arnej27959 opened the pull request:
* change the ReferenceOperations peek API also, putting
  the input parameter as the first child, and numbering
  child indexes in the Spec starting at 1.

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@havardpe please review

havardpe commented on eval/src/vespa/eval/eval/test/reference_evaluation.cpp:
```diff
@@ -135,7 +130,7 @@ struct EvalNode : public NodeVisitor {
             }
         }
-        result = ReferenceOperations::peek(param, spec, children).normalize();
+        result = ReferenceOperations::peek(spec, children).normalize();
```
should no longer normalize here, since it is done in eval_node instead

arnej27959 replied:
fixed by rebasing and resolving conflicts

havardpe commented on eval/src/vespa/eval/eval/test/reference_operations.cpp:
```diff
@@ -175,14 +175,15 @@ TensorSpec ReferenceOperations::merge(const TensorSpec &a, const TensorSpec &b,
 
-TensorSpec ReferenceOperations::peek(const TensorSpec &param, const PeekSpec &peek_spec, const std::vector<TensorSpec> &children) {
+TensorSpec ReferenceOperations::peek(const PeekSpec &peek_spec, const std::vector<TensorSpec> &children) {
     if (peek_spec.empty()) {
```
should also check for empty children

arnej27959 replied:
fixed

arnej27959 committed:
also check for empty children

arnej27959 committed:
renumber Peek children, the sequel

* change the ReferenceOperations peek API also, putting
  the input parameter as the first child, and numbering
  child indexes in the Spec starting at 1.
